pkg/infinity/oauth.go

```diff
@@ -111,13 +113,18 @@ func ApplyAWSAuth(ctx context.Context, httpClient *http.Client, settings models.
 			AccessKey: settings.AWSAccessKey,
 			SecretKey: settings.AWSSecretKey,
 		}
-		rt, _ := sigv4.New(conf, sigv4.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+		authSettings, found := awsds.ReadAuthSettingsFromContext(ctx)
```

Comment:
@zoltanbedi do you know why do we need to add this and we didn't before? Was this previously hidden in `sigv4.New`? Do you know if we are always going to have those settings in context?

Reply:
yeah so the original dependabot PR fails because the `"github.com/grafana/grafana-aws-sdk/pkg/awsds"` package is using an older version of plugin sdk and they are not compatible. I bumped the aws sdk and it had this breaking change in there. I'm not sure if it will always have the settings but that's why it returns the found variable.